client/src/main/scala/spatutorial/client/components/TerminalUserDeskRecs.scala

```diff
@@ -157,7 +166,7 @@ object TableTerminalDeskRecs {
         val fill = item.queueDetails.flatMap(
           (q: QueueDetailsRow) => {
             val warningClasses = if (q.waitTimeWithCrunchDeskRec < q.waitTimeWithUserDeskRec) "table-warning" else ""
-            val dangerWait = if (q.waitTimeWithUserDeskRec > slaFromTerminalAndQueue(p.terminalName, q.queueName)) "table-danger"
+            val dangerWait = if (q.waitTimeWithUserDeskRec > p.airportConfigHolder.slaByQueue(q.queueName)) "table-danger"
```

Comment:
this component is per queue, right? 
We should avoid passing in complex structures which we then pluck bits out of. Do the slaByQueue at some higher level, at the same point we've bound p.queueName